awx/main/tests/functional/api/test_workflow_node.py

```diff
@@ -56,6 +68,90 @@ def test_node_accepts_prompted_fields(inventory, project, workflow_job_template,
          user=admin_user, expect=201)
 
 
+@pytest.mark.django_db
+class TestApprovalNodes():
+    def test_approval_node_creation(self, post, approval_node, admin_user):
+        url = reverse('api:workflow_job_template_node_create_approval',
+                      kwargs={'pk': approval_node.pk, 'version': 'v2'})
+        post(url, {'name': 'Test', 'description': 'Approval Node', 'timeout': 0},
+             user=admin_user, expect=200)
+
+        approval_node = WorkflowJobTemplateNode.objects.get(pk=approval_node.pk)
+        assert isinstance(approval_node.unified_job_template, WorkflowApprovalTemplate)
+        assert approval_node.unified_job_template.name=='Test'
+        assert approval_node.unified_job_template.description=='Approval Node'
+        assert approval_node.unified_job_template.timeout==0
+
+    def test_approval_node_creation_failure(self, post, approval_node, admin_user):
+        url = reverse('api:workflow_job_template_node_create_approval',
+                      kwargs={'pk': approval_node.pk, 'version': 'v2'})
+        post(url, {'name': '', 'description': 'Approval Node', 'timeout': 0},
+             user=admin_user, expect=400)
+        # Leave off a required param to assert that you get a 400
+        approval_node = WorkflowJobTemplateNode.objects.get(pk=approval_node.pk)
+        assert isinstance(approval_node.unified_job_template, WorkflowApprovalTemplate) is False
+
+    @pytest.mark.parametrize("is_admin, is_org_admin, status", [
+        [True, False, 200],
```

Comment:
Some comments at the end of each of these lines might be helpful e.g.,

# if they're an WFJT admin, they get a 200
# if they're not an WFJT *or* org admin, they get a 403
# if they're an organization admin, they get a 200